lib/driver.rb

```diff
@@ -0,0 +1,91 @@
+require_relative 'csv_record'
+FEE = 1.65
+DRIVER_COMMISSION = 0.8
+
+module RideShare
+  class Driver < CsvRecord
+    attr_reader :name, :vin, :trips
+    attr_accessor :status
+
+    def initialize(id:, name:, vin:, status: :AVAILABLE , trips: [])
+      super(id)
+
+      @name = name
+      @vin = vin
+      @status = status.to_sym
+      @trips = trips
+
+      unless @vin.size == 17
+        raise ArgumentError.new("#{@vin} is not the right size")
+      end
+
+      unless @status == :AVAILABLE || @status == :UNAVAILABLE
+        raise ArgumentError.new("Invalid status")
+      end
+
+    end
+
+    #TO-DO: create add_trip method
+    def add_trip(trip)
+      @trips << trip
+    end
+
+    # driver helper method to change driver status to unavailable
+    def change_status
+      @status = :UNAVAILABLE
+    end
+
+
+    def average_rating
+      # need to account if rating is nil
+      count = 0
+      rating_sum = 0
+      if @trips.empty?
+        return 0
+      else
+        @trips.each do |trip|
+          if trip.end_time == nil
+            rating_sum += 0
+          elsif
+          rating_sum += trip.rating
+            count += 1
+          end
```

Comment:
You can remove the incomplete trips with `reject`.

```suggestion
        @trips.reject { |trip| trip.end_time.nil? }.each do |trip|
          rating_sum += trip.rating
          count += 1
```